h2/src/main/org/h2/mode/FunctionsPostgreSQL.java

```diff
@@ -235,6 +247,13 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             // Not implemented
             result = v1.convertTo(Value.VARCHAR);
             break;
+        case ARRAY_TO_STRING:
+            StringJoiner joiner = new StringJoiner(String.valueOf(v1.getObject()));
+            for (Value v : ((ValueArray) v0).getList()) {
+                joiner.add(String.valueOf(v.getObject()));
+            }
```

Comment:
Don't use `Value.getObject()` in functions, objects returned by it aren't required to return something useful from their `toString()` methods.

For this specific function `getString()` is a more reasonable method.

This function also needs some tests.